googlemock/CMakeLists.txt

```diff
@@ -106,24 +106,19 @@ else()
   target_link_libraries(gmock_main gmock)
 endif()
 
-# If the CMake version supports it, attach header directory information
-# to the targets for when we are part of a parent build (ie being pulled
-# in via add_subdirectory() rather than being a standalone build).
-if (DEFINED CMAKE_VERSION AND NOT "${CMAKE_VERSION}" VERSION_LESS "2.8.11")
-  target_include_directories(gmock      SYSTEM INTERFACE "${gmock_SOURCE_DIR}/include")
-  target_include_directories(gmock_main SYSTEM INTERFACE "${gmock_SOURCE_DIR}/include")
-endif()
-
 ########################################################################
 #
 # Install rules
 if(INSTALL_GMOCK)
   install(TARGETS gmock gmock_main
+    EXPORT GMockTargets
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
```

Comment:
Ack. Added version checking. But please, notice that due to this switching logic, I preferred to use the `set_target_properties(INTERFACE_INCLUDE_DIRECTORIES)` so as to avoid repeating the full `install()` command in the two branches of the `if` statement.